armi/reactor/tests/test_hexBlockRotate.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2024 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the ability to rotate a hexagonal block."""
+
+import copy
+import math
+import unittest
+
+import numpy as np
+
+from armi.reactor.blocks import HexBlock
+from armi.reactor.components import Component
+from armi.reactor.grids import (
+    MultiIndexLocation,
+    CoordinateLocation,
+    IndexLocation,
+    HexGrid,
+)
+from armi.reactor.tests.test_blocks import loadTestBlock, NUM_PINS_IN_TEST_BLOCK
+from armi.utils import iterables
+
+
+class HexBlockRotateTests(unittest.TestCase):
+    """Tests for various rotation aspects of a hex block."""
+
+    BOUNDARY_PARAMS = [
+        "cornerFastFlux",
+        "pointsCornerDpa",
+        "pointsCornerDpaRate",
+        "pointsCornerFastFluxFr",
+        "pointsEdgeDpa",
+        "pointsEdgeDpaRate",
+        "pointsEdgeFastFluxFr",
+        "THedgeTemp",
+        "THcornTemp",
+    ]
+    BOUNDARY_DATA = np.arange(6, dtype=float) * 10
+
+    PIN_PARAMS = [
+        "percentBuByPin",
+        "linPowByPin",
+    ]
+
+    PIN_DATA = np.arange(NUM_PINS_IN_TEST_BLOCK, dtype=float)
+
+    @classmethod
+    def setUpClass(cls):
+        cls.BASE_BLOCK = loadTestBlock()
+        cls._assignParamData(cls.BOUNDARY_PARAMS, cls.BOUNDARY_DATA)
+        cls._assignParamData(cls.PIN_PARAMS, cls.PIN_DATA)
+
+    @classmethod
+    def _assignParamData(cls, names: list[str], referenceData: np.ndarray):
+        """Assign initial rotatable pararameter data.
+
+        Make some arrays, some lists to make sure we have good coverage of usage.
+        """
+        # Yes we're putting the variable type in the name but that's why this method exists
+        listData = referenceData.tolist()
+        for ix, name in enumerate(names):
+            cls.BASE_BLOCK.p[name] = referenceData if (ix % 2) else listData
+
+    def test_orientationVector(self):
+        """Test the z-value in the orientation vector matches rotation.
+
+        .. test:: Demonstrate that a HexBlock can be rotated in 60 degree increments.
+            :id: T_ARMI_ROTATE_HEX
+            :tests: R_ARMI_ROTATE_HEX
+
+        .. test:: After rotating a block, the orientation parameter reflects the current rotation.
+            :id: T_ARMI_ROTATE_HEX_ORIENTATION
+            :tests: R_ARMI_ROTATE_HEX_PARAMS
+            :acceptance_criteria: Through a unit test, compare the z-rotation entry against the rotation amount.
```

Comment:
```suggestion
            :acceptance_criteria: Compare the z-rotation entry against the rotation amount.
```

Reply:
We can't have the `acceptance_criteria` for every single ARMI req start off with "in a unit test", it's redundant.

IMHO